src/main/java/com/google/devtools/build/lib/analysis/starlark/UnresolvedSymlinkAction.java

```diff
@@ -81,6 +82,16 @@ public ActionResult execute(ActionExecutionContext actionExecutionContext)
       throw new ActionExecutionException(message, e, this, false, code);
     }
 
+    // Action filesystems are responsible for their own metadata injection.
+    if (actionExecutionContext.getActionFileSystem() == null) {
+      actionExecutionContext
```

Comment:
I am wondering whether this can be done inside skyframe (i.e. we don't need to add special case for action filesystem)?

cc @tjgq